src/config.rs

```diff
@@ -13,39 +13,48 @@
 
 use crate::{util::ReadableSize, Result};
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum RecoveryMode {
+    TolerateCorruptedTailRecords = 0,
+    AbsoluteConsistency = 1,
+}
+
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct Config {
     pub dir: String,
-    pub recovery_mode: i32,
+    pub recovery_mode: RecoveryMode,
     pub bytes_per_sync: ReadableSize,
     pub target_file_size: ReadableSize,
-    pub cache_size_limit: ReadableSize,
-    pub total_size_limit: ReadableSize,
 
-    // Use raftstore.cfg.raft_log_gc_threshold
-    #[doc(hidden)]
-    #[serde(skip_serializing)]
-    pub compact_threshold: usize,
+    /// Only purge if disk file size is greater than `purge_threshold`.
+    pub purge_threshold: ReadableSize,
+
+    /// Only purge if garbage is greater than `garbage_threshold`.
+    pub garbage_threshold: f64,
+
+    /// Total size limit to cache log entries.
+    ///
+    /// FIXME: it doesn't make effect currently.
+    pub cache_limit: ReadableSize,
 
-    // Use raftstore.cfg.region_split_size
-    #[doc(hidden)]
-    #[serde(skip_serializing)]
-    pub region_size: ReadableSize,
+    /// Size limit to cache log entries for every Raft.
+    pub cache_limit_per_raft: ReadableSize,
 }
 
 impl Default for Config {
     fn default() -> Config {
         Config {
             dir: "".to_owned(),
-            recovery_mode: 0,
+            recovery_mode: RecoveryMode::TolerateCorruptedTailRecords,
             bytes_per_sync: ReadableSize::kb(256),
             target_file_size: ReadableSize::mb(128),
-            cache_size_limit: ReadableSize::gb(2),
-            total_size_limit: ReadableSize::gb(20),
-            compact_threshold: 0,
-            region_size: ReadableSize::mb(0),
+            purge_threshold: ReadableSize::gb(2),
+            garbage_threshold: 0.75,
+            cache_limit: ReadableSize::gb(1),
+            cache_limit_per_raft: ReadableSize::mb(4),
```

Comment:
I think the default value is a bit of small, please think about the scenario that there is a very little number of hot regions.